src/main/java/io/confluent/connect/jdbc/TimestampIncrementingTableQuerier.java

```diff
@@ -144,21 +148,30 @@ protected void createPreparedStatement(Connection db) throws SQLException {
     stmt = db.prepareStatement(queryString);
   }
 
+
+
   @Override
   protected ResultSet executeQuery() throws SQLException {
     if (incrementingColumn != null && timestampColumn != null) {
-      Timestamp ts = new Timestamp(timestampOffset == null ? 0 : timestampOffset);
-      stmt.setTimestamp(1, ts, UTC_CALENDAR);
-      stmt.setLong(2, (incrementingOffset == null ? -1 : incrementingOffset));
-      stmt.setTimestamp(3, ts, UTC_CALENDAR);
-      log.debug("Executing prepared statement with timestamp value = " + timestampOffset + " and incrementing value = " + incrementingOffset);
+      Timestamp startTime = new Timestamp(timestampOffset == null ? 0 : timestampOffset);
+      Timestamp endTime = new Timestamp(JdbcUtils.getCurrentTimeOnDB(stmt.getConnection(), UTC_CALENDAR).getTime() - timestampDelay);
+      stmt.setTimestamp(1, endTime, UTC_CALENDAR);
+      stmt.setTimestamp(2, startTime, UTC_CALENDAR);
+      stmt.setLong(3, (incrementingOffset == null ? -1 : incrementingOffset));
+      stmt.setTimestamp(4, startTime, UTC_CALENDAR);
+      log.debug("Executing prepared statement with start time value = " + timestampOffset + " (" + startTime.toString() + ") "
+              + " end time " + endTime.toString()
+              + " and incrementing value = " + incrementingOffset);
     } else if (incrementingColumn != null) {
       stmt.setLong(1, (incrementingOffset == null ? -1 : incrementingOffset));
       log.debug("Executing prepared statement with incrementing value = " + incrementingOffset);
     } else if (timestampColumn != null) {
-      Timestamp ts = new Timestamp(timestampOffset == null ? 0 : timestampOffset);
-      stmt.setTimestamp(1, ts, UTC_CALENDAR);
-      log.debug("Executing prepared statement with timestamp value = " + timestampOffset);
+      Timestamp startTime = new Timestamp(timestampOffset == null ? 0 : timestampOffset);
+      Timestamp endTime = new Timestamp(new Date().getTime() - timestampDelay);
```

Comment:
Missed updating this one -- should use `JdbcUtils.getCurrentTimeOnDB()` like the `timestamp+increasing` mode.